product-service/handlers/catalog-batch-process/index.ts

```diff
@@ -0,0 +1,52 @@
+import SNS from 'aws-sdk/clients/sns';
+import { SQSHandler } from 'aws-lambda';
+import { Client } from 'pg';
+import dbOptions from '../../utils/dbOptions';
+
+export const handler: SQSHandler = async (event) => {
+  console.log("Catalog Batch process handler invoked");
+  const { SNS_ARN } = process.env;
+  let client = null;
+
+  try {
+    const sns = new SNS();
+    client = new Client(dbOptions);
+    await client.connect();
+
+    for (const item of event.Records) {
+      const { title, description, price } = JSON.parse(item.body);
+      const parsedPrice = Number(price);
+
+      if (
+        typeof title !== "string" ||
+        typeof description !== "string" ||
+        typeof parsedPrice !== "number"
+      ) {
+        throw { statusCode: 400, message: 'Product data is invalid' };
```

Comment:
Keep in mind that you may get more than one record in this lambda, and only one of the records may contain invalid data. 
If you throw an error here all other records (valid) won't be saved into DB.